test/extended/router/h2spec.go

```diff
@@ -153,63 +216,43 @@ func failingTests(testSuites []*h2spec.JUnitTestSuite) []h2specFailingTest {
 	return failures
 }
 
-func runConformanceTests(oc *exutil.CLI, t h2specRouteTypeTest) []*h2spec.JUnitTestSuite {
-	podName := "h2spec"
-	e2e.ExpectNoError(e2epod.WaitForPodNameRunningInNamespace(oc.KubeClient(), podName, oc.KubeFramework().Namespace.Name))
-
-	var results []*h2spec.JUnitTestSuite
+func runConformanceTests(oc *exutil.CLI, host, podName string) ([]*h2spec.JUnitTestSuite, error) {
+	g.By("Running the h2spec CLI test")
 
-	o.Expect(wait.PollImmediate(time.Second, 5*time.Minute, func() (bool, error) {
-		g.By("Running the h2spec CLI test")
+	// this is the output file in the pod
+	outputFile := "/tmp/h2spec-results"
 
-		ns := oc.KubeFramework().Namespace.Name
-		outputFile, err := ioutil.TempFile("", "runConformanceTests")
-		if err != nil {
-			e2e.Logf("[%s] error: failed to generate tmp file: %v; retrying...", t.routeType, err)
-			return false, nil
-		}
-		defer os.Remove(outputFile.Name())
-
-		output, _ := e2e.RunHostCmd(ns, podName, h2specCommand(h2specDialTimeoutInSeconds, t.hostname, outputFile.Name()))
-		// h2spec will exit with non-zero if _any_ test in the
-		// suite fails, or if there is a dial timeout, so we
-		// ignore the error. We can exit from this loop if we
-		// can fetch the results, if we can decode the results
-		// and we have > 0 test suites from the decoded
-		// results.
-
-		g.By(fmt.Sprintf("[%s] Copying results from %s:%q", t.routeType, podName, outputFile.Name()))
-		data, err := e2e.RunHostCmd(ns, podName, fmt.Sprintf("cat %q", outputFile.Name()))
-		if err != nil {
-			e2e.Logf("[%s] error: failed to copy results: %v; retrying...", t.routeType, err)
-			return false, nil
-		}
-		if data == "" {
-			e2e.Logf("[%s] error: zero length results file; retrying...", t.routeType)
-			return false, nil
-		}
+	// h2spec will exit with non-zero if _any_ test in the suite
+	// fails, or if there is a dial timeout, so we ignore the
+	// error. If we can fetch the results and if we can decode the
+	// results and we have > 0 test suites from the decoded
+	// results then assume the test ran.
+	output, _ := e2e.RunHostCmd(oc.Namespace(), podName, h2specCommand(h2specDialTimeoutInSeconds, host, outputFile))
```

Comment:
Here is another section which could use gomega too

Reply:
The behaviour here is now different; we wait for DNS to register for the route under test. Once DNS has registered successfully we now expect the test to run. There should no longer be any need to retry.